src/orion/core/io/cmdline_parser.py

```diff
@@ -20,28 +20,39 @@
 
 
 class CmdlineParser(object):
+    """CmdlineParser is aimed at providing a simple tool to interpret commandline arguments
+    for the purposes of Orion. It can transform a list of string representing arguments to their
+    corresponding values. It can also recreate that string from the values by maintaing a template
+    of the way the arguments were passed.
+    """
+
     def __init__(self):
+        """Initialize the OrderedDict of arguments and the template list"""
         self.arguments = OrderedDict()
         self._already_parsed = False
         self.template = []
 
     def format(self, configuration):
+        """Recreate the string of argument using the template made at parse-time.
+        The `configuration` argument must be a `dict` compatible with the template.
+        """
```

Comment:
Parameters shoud be document as described [here](https://docs.scipy.org/doc/numpy/docs/howto_document.html#sections)

Reply:
Also, same comment for one-line summary followed by description.